### DISCUSSION TITLE
Having issues with integration turning to Failed To Setup every other day

### DISCUSSION THREAD

**Discussion by boxofrabbits** [BottlecapDave/HomeAssistant-OctopusEnergy#603] [General]
Hey guys. I've been trying to get the integration up and running, but have had no luck.
The actual integration seems to show information well, but every other day or so it all stops working. The integration displays "Failed to setup" and I have to reload it to get it working again, but this only works for a day or so. 
I've also had zero luck getting the info to display in the HA Energy Dashboard (I've followed the Energy Dashboard setup instructions carefully and twice with a full reset inbetween)
Can't work out what I'm doing wrong. 

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/75987600/9b7767ad-ecb8-4132-9336-9de55eb1af09)

**Comment by boxofrabbits** [comment]
Have found this in the log

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/75987600/4ba74b8a-b58d-4b09-a1fb-b0861e6ea461)

**Reply by JRascagneres** [reply]
This looks like an Octopus API issue. I'm now hitting it too and have been setup fine for months

**Reply by JRascagneres** [reply]
I seriously regret reloading the integration though

**Reply by BottlecapDave** [reply]
This API call will only call when setting up the integration (e.g. on reload or on HA restart). I'll add it to my list to combine this call with the account info to try and reduce the chance of failure. 

However this wouldnt stop the integration running every few days (unless you're restarting your HA instance every day and hitting the above issue). Do you have any other errors in your logs?

**Reply by terencekmchan** [reply]
i have the same issue this morning.

I used a curl call to query the API inside the terminal from my HA and it seems to work fine.

I am getting 502 bad gateway from the log. 

Anything i can do to troubleshoot??

**Reply by BottlecapDave** [reply]
A 502 is a server error on their side, so very little we can do.

**Reply by terencekmchan** [reply]
This is weird. I wonder why was i able to call the Octopus api using curl from the same machine.

Is there anyway I can find out what the actual API is doing and I can try to run them inside the terminal ? 

<img width="880" alt="Screenshot 2023-12-05 at 7 00 57 AM" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/116647715/9dcef94e-650f-46c1-919a-44344d2ebed3">

<img width="1691" alt="Screenshot 2023-12-05 at 6 58 59 AM" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/116647715/f14b3d86-3a89-4a81-9e2d-49f8c815622d">

**Reply by BottlecapDave** [reply]
It's not necessarily the whole API, it could be a backend service that powers the graphql account request

**Reply by gwilford** [reply]
I'm also getting 502 bad gateway after reloading this morning. No data updates in HA since around 02:45.

**Reply by boxofrabbits** [reply]
This popped up at 6:11 this morning

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/75987600/ec3ee221-e5b7-430c-a6fa-a64bdd6aa29a)

**Reply by boxofrabbits** [reply]
![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/75987600/92878092-4a6f-4624-aca7-f300f8447dfb)

**Reply by BottlecapDave** [reply]
This was due to an OE API outage

**Reply by brutus6** [reply]
Yes. I am getting this as well. All my Octopus sensors are not recording. Is it a case of waiting for OE API to be back up and running or does it require any action from the user in HA?

**Reply by JRascagneres** [reply]
You just have to wait for it to be up.

**Reply by brutus6** [reply]
Okay. I will wait and see if they just start working again.

**Reply by brutus6** [reply]
Update: It is working for me now.

**Comment by boxofrabbits** [comment]
Ok, but I've been having this trouble for weeks. I've tried my best to fix
it myself, but no luck.

**Reply by JRascagneres** [reply]
I mean its back up now, try it. If it doesn't work now then there are bigger issues but its fine for me now and has been fine for months.

**Reply by brutus6** [reply]
Hmmm.... Mine has been fine for a while, but stoped last night some time and hasn't worked all day. I tried reconfiguring and checked the API key all correct. Will just wait and see if the API goes back online and see if that fixes it. Just wasn't sure if there was some sort of refresh.

**Reply by JRascagneres** [reply]
Should be able to head into integrations and reload it.

**Reply by brutus6** [reply]
Scrap that we are back in business. Must have come back on line between now and when I last did that. Thanks!
